atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/progress/DefaultBucketProgressStoreTest.java

```diff
@@ -0,0 +1,176 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.asts.progress;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetRequest;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.sweep.asts.SweepStateCoordinator.SweepableBucket;
+import com.palantir.atlasdb.sweep.queue.ShardAndStrategy;
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+// Much of the tests of basic behaviour on this class are covered by the abstract test, because we are concerned
+// about how we actually work with the underlying behaviour of the key-value-services (and, some of that functionality
+// was added as part of this work).
+@ExtendWith(MockitoExtension.class)
+public class DefaultBucketProgressStoreTest {
+    private static final RuntimeException GENERIC_RUNTIME_EXCEPTION = new RuntimeException("oops");
+    private static final SweepableBucket DEFAULT_BUCKET = SweepableBucket.of(ShardAndStrategy.thorough(0), 1L);
+    private static final Cell DEFAULT_BUCKET_CELL = DefaultBucketKeySerializer.INSTANCE.bucketToCell(DEFAULT_BUCKET);
+
+    private static final BucketProgressSerializer BUCKET_PROGRESS_SERIALIZER =
+            BucketProgressSerializer.create(ObjectMappers.newSmileServerObjectMapper());
+
+    @Mock
+    private KeyValueService keyValueService;
+
+    private BucketProgressStore bucketProgressStore;
+
+    @BeforeEach
+    public void setUp() {
```

Comment:
do you want setUp or setup? You've two different names in the same PR. That said, we're horribly inconsistent about this in the codebase - there's before, initialise, setup, setUp.

Reply:
I'll pick the one that's common to the location of the tests and stick to that.

Reply:
yeah `setUp` seems most common I think